docs/conf.py

```diff
@@ -0,0 +1,99 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# This file only contains a selection of the most common options. For a full
+# list see the documentation:
+# http://www.sphinx-doc.org/en/master/config
+
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+import subprocess, os
+
+def configureDoxyfile(input_dir, output_dir, doxyfile_in, doxyfile_out):
+
+	with open(doxyfile_in, 'r') as file :
+		filedata = file.read()
+
+	filedata = filedata.replace('@CMAKE_SOURCE_DIR@', input_dir)
+	filedata = filedata.replace('@DOXYGEN_OUTPUT_DIR@', output_dir)
+	
+	with open(doxyfile_out, 'w') as file:
+		file.write(filedata)
+
+# -- Project information -----------------------------------------------------
+
+project = 'KokkosFFT'
+copyright = '2024, Yuuichi Asahi'
+author = 'Yuuichi Asahi'
+
+# Check if we're running on Read the Docs' servers
+read_the_docs_build = os.environ.get('READTHEDOCS', None) == 'True'
+
+breathe_projects = {}
+
+if read_the_docs_build:
+    cwd = os.getcwd()
+    print(cwd)
+
+    # Update submodule
+    #update_submodule = ['git', 'submodule', 'update', '--init', '--recursive']
+    #subprocess.call(update_submodule, shell=True)
+
+    # Build Docs with CMake 
+    ###cmake_commands = 'cmake -DBUILD_TESTING=OFF \
+    ###                        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    ###                        -DKokkos_ENABLE_OPENMP=ON \
+    ###                        -DBUILD_EXAMPLES=OFF \
+    ###                        -DKokkosFFT_INTERNAL_Kokkos=ON \
+    ###                        -DKokkosFFT_ENABLE_DOCS=ON ..'
+    ###build = 'cmake --build . -j 2'
+    ###                        
+    ###subprocess.call(f'cd ../; mkdir build; cd build; {cmake_commands}; {build}', shell=True)
+    input_dir = cwd + '/..'
+    output_dir = cwd +'/doxygen/'
+    doxyfile_in = cwd + '/Doxyfile.in'
+    doxyfile_out = cwd + '/Doxyfile'
+    configureDoxyfile(input_dir, output_dir, doxyfile_in, doxyfile_out)
+    subprocess.call('pwd; ls -lat; doxygen Doxyfile; ls -lat doxygen/xml', shell=True)

```

Comment:
here again I don't understand, does CMake handle Doxygen call or python?

Reply:
In the rtd workflow, only the python works. CMake is not used.